packages/headless/src/controllers/folding/headless-folding.ts

```diff
@@ -0,0 +1,125 @@
+import {NumberValue, Schema, StringValue} from '@coveo/bueno';
+import {Engine} from '../../app/engine';
+import {search, configuration, folding} from '../../app/reducers';
+import {registerFolding} from '../../features/folding/folding-actions';
+import {FoldedResult} from '../../features/folding/folding-state';
+import {
+  ConfigurationSection,
+  FoldingSection,
+  SearchSection,
+} from '../../state/state-sections';
+import {loadReducerError} from '../../utils/errors';
+import {validateOptions} from '../../utils/validate-payload';
+import {buildController, Controller} from '../controller/headless-controller';
+
+export {FoldedResult};
+
+const optionsSchema = new Schema<Required<FoldingOptions>>({
+  collectionField: new StringValue(),
+  parentField: new StringValue(),
+  childField: new StringValue(),
+  maximumFoldedResults: new NumberValue(),
+});
+
+export interface FoldingOptions {
+  /**
+   * The name of the field used to resolve what collection a result is part of.
+   *
+   * @defaultValue `foldedcollection`
+   */
+  collectionField?: string;
+  /**
+   * The name of the field used to reference which result is the parent of another one within a collection.
+   *
+   * @defaultValue `foldedparent`
+   */
+  parentField?: string;
+  /**
+   * The name of the field used to uniquely identify a result within a collection.
+   *
+   * @defaultValue `foldedchild`
+   */
+  childField?: string;
+  /**
+   * The maximum number of additional results to fetch from the same collection
```

Comment:
I'm not sure I understand this. Is this parameter supposed to be like the following in jsui: https://coveo.github.io/search-ui/components/folding.html#options.maximumexpandedresults ?

Reply:
it's the equivalent of JSUI's [`range`](https://coveo.github.io/search-ui/components/folding.html#options.range)

Reply:
Ok then I would recommend the following:
`The number of child results to fold under the root collection element, before expansion.`